core/trino-main/src/main/java/io/trino/sql/planner/NodePartitioningManager.java

```diff
@@ -260,16 +263,27 @@ public BucketNodeMap getBucketNodeMap(Session session, PartitioningHandle partit
      */
     private int getDefaultBucketCount(Session session, PartitioningHandle partitioningHandle)
     {
-        // The goal is to have enough buckets to evenly distribute them across tasks created.  The simplest way to do that is to have exactly as many buckets as there are tasks;
-        // then each task gets one bucket.  So let's return the total number of tasks here.  This number, unfortunately, is hard to know precisely because of the variety of ways in
-        // which schedulers create tasks.  We therefore have to estimate.
+        // The default bucket count is used by both remote and local exchanges to assign buckets to nodes and drivers. The goal is to have enough
+        // buckets to evenly distribute them across tasks or drivers. If number of buckets is too low, then some tasks or drivers will be idle.
+        // Excessive number of buckets would make bucket lists or arrays too large.
+
+        // For remote exchanges bucket count can be assumed to be equal to node count multiplied by some constant.
```

Comment:
"by some constant"? really - please explain why we are bumping that over node count even though we expect number of tasks to be equal to node count or less